src/main/java/core/basesyntax/FileWork.java

```diff
@@ -1,7 +1,39 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.regex.Pattern;
+
 public class FileWork {
     public String[] readFromFile(String fileName) {
-        return null;
+        String[] checkEmpty = new String[0];
+        StringBuilder selectedWords = new StringBuilder();
+
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fileName))) {
+            String line = bufferedReader.readLine();
+            Pattern pattern = Pattern.compile("[ ,.!?:;]");
+
+            while (line != null) {
+                line = line.toLowerCase();
+                String[] arrayOfWordsOfLine = pattern.split(line);
+                for (String wordOfLine : arrayOfWordsOfLine) {
+                    if (wordOfLine.startsWith("w")) {
```

Comment:
this comment is not fixed, make "w" and your regex constants

Reply:
Ouch!  I'm sorry! I did't cath that! 
done